errors/truncate_error.go

```diff
@@ -0,0 +1,71 @@
+package errors
+
+import (
+	"github.com/cashapp/blip"
+)
+
+type TruncateErrorPolicy struct {
+	Policy
+	hadTruncateError bool
+}
+
+func NewTruncateErrorPolicy(s string) *TruncateErrorPolicy {
+	p := NewPolicy(s)
+	return &TruncateErrorPolicy{
+		Policy: *p,
```

Comment:
No sure this is right because `NewPolicy(s string) *Policy`, so this would be `**p`, I think?

Reply:
Or, maybe it's somehow needed/due to embedding?

Reply:
I'm using it as a value rather than a reference, but the method returns a pointer, so I have to dereference it to get the value to assign it to the field. It passes both the unit tests and integration testing so I think we are good.

Reply:
Ah right, I see what's going on. Idk why but I was thinking `**p` as pointer to pointer instead of Go deref.